From c44552f8413dd5f385caf70d4e30669ea24564ef Mon Sep 17 00:00:00 2001
From: wangjianxin <wangjianxin@canaan-creative.com>
Date: Wed, 5 Jun 2024 19:17:10 +0800
Subject: [PATCH 01/20] canaan soc modify

---
 arch/riscv/Kconfig.socs | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/arch/riscv/Kconfig.socs b/arch/riscv/Kconfig.socs
index 6833d01e2..f04937388 100644
--- a/arch/riscv/Kconfig.socs
+++ b/arch/riscv/Kconfig.socs
@@ -66,11 +66,17 @@ config SOC_VIRT
 	  This enables support for QEMU Virt Machine.
 
 config ARCH_CANAAN
-	def_bool SOC_CANAAN
+	bool "Canaan Kendryte SoC"
+	help
+	  This enables support for Canaan Kendryte series SoC platform hardware.
 
 config SOC_CANAAN
+	def_bool SOC_CANAAN_K210
+	depends on ARCH_CANAAN
+
+config SOC_CANAAN_K210
 	bool "Canaan Kendryte K210 SoC"
-	depends on !MMU
+	depends on !MMU && ARCH_CANAAN
 	select CLINT_TIMER if RISCV_M_MODE
 	select ARCH_HAS_RESET_CONTROLLER
 	select PINCTRL
@@ -78,7 +84,7 @@ config SOC_CANAAN
 	help
 	  This enables support for Canaan Kendryte K210 SoC platform hardware.
 
-if ARCH_CANAAN
+if SOC_CANAAN_K210
 
 config ARCH_CANAAN_K210_DTB_BUILTIN
 	def_bool SOC_CANAAN_K210_DTB_BUILTIN
-- 
2.39.1

